tests/test_options.py

```diff
@@ -50,6 +51,11 @@ def test_can_set_service_name_with_envvar(monkeypatch):
     options = HoneycombOptions()
     assert options.service_name == "my-service"
 
+def test_can_set_generic_api_endpoint_with_envvar(monkeypatch):
```

Comment:
Nice! I think we should add a test for setting with param as well

Reply:
I believe it's already covered by what was written already (the tests failed when I implemented the default vs param vs envvar fallback wrong on the first try?)

I can duplicate the coverage and make it a more explicit test tho!

Reply:
Thank you! Yeah I think it was covered in traces and metrics endpoint, but appreciate the extra explicit one 👍